### ISSUE TITLE
Regression, GeoTransform missing from NetCDF file in GDAL-current

### ISSUE THREAD

**Issue by mmomtchev** [OSGeo/gdal#4581]
## Steps to reproduce the problem.

No GeoTransform is reported for the following file in GDAL built from a git checkout:
```
gdalinfo NETCDF:"/vsis3/noaa-gfs-bdp-pds/gfs.20210930/06/atmos/gfs.t06z.atmf012.nc":tmp
```
GDAL 3.2.3 does report a GeoTransform (ok)
GDAL 3.3.2 does not either (nok)

## Operating system

Ubuntu 20.04 64 bit

## GDAL version and provenance

Build with `./configure --with-python=yes`
GDAL-current git checkout
GDAL/release/3.2 git checkout
GDAL/release/3.3 git checkout

**Comment by rouault** [comment]
The fix might be similar to 493c57e665b7b8ed2adc0d0bdf5cfacc0c84af05 to add the attribute name & values used by this dataset. Since 3.3.2 we have hardened the way to identify dimension axis to be recognized as geospatial ones

**Comment by mmomtchev** [comment]
`GDAL_NETCDF_IGNORE_XY_AXIS_NAME_CHECKS=YES` fixes the problem
